core/src/rpc/rpc_context.rs

```diff
@@ -481,14 +483,55 @@ pub trait RpcContext {
 	}
 
 	// ------------------------------
-	// Methods for querying
+	// Methods for relating
 	// ------------------------------
 
 	async fn relate(&self, _params: Array) -> Result<impl Into<Data>, RpcError> {
 		let out: Result<Value, RpcError> = Err(RpcError::MethodNotFound);
 		out
 	}
 
+	// ------------------------------
+	// Methods for running functions
+	// ------------------------------
+
+	async fn run(&self, params: Array) -> Result<impl Into<Data>, RpcError> {
+		let Ok((Value::Strand(Strand(func_name)), version, args)) = params.needs_one_two_or_three()
+		else {
+			return Err(RpcError::InvalidParams);
+		};
+
+		let version = match version {
+			Value::Strand(Strand(v)) => Some(v),
+			Value::None | Value::Null => None,
+			_ => return Err(RpcError::InvalidParams),
+		};
+
+		let args = match args {
+			Value::Array(Array(arr)) => arr,
+			_ => vec![],
```

Comment:
Should we perhaps throw on an invalid argument? It's such a shame that we need to keep `null` in for the sake of JSON 😒
```suggestion
			Value::None | Value::Null => vec![],
			_ => return Err(RpcError::InvalidParams),
```